lib/service/database.js

```diff
@@ -48,6 +48,21 @@ DatabaseService.prototype.run = function(query, callback) {
     this.queryRunner.run(query, QUERY_RUNNER_READONLY_OP, callback);
 };
 
+DatabaseService.prototype.run_writable = function(query, callback) {
+    this.queryRunner.run(query, QUERY_RUNNER_WRITE_OP, callback);
+};
+
+var TIMEOUT_ERROR_MESSAGE = new RegExp('canceling statement due to statement timeout', 'i');
+
+// check err returned by queryRunner operation for a timeout error
+DatabaseService.prototype.checkForTimeout = function(err) {
+    if (err && err.message && err.message.match(TIMEOUT_ERROR_MESSAGE)) {
```

Comment:
I did this to avoid returning anything other than `true` or `false` (the regular expression match). Do you think it is OK to return a truthy value different from true.

Otherwise I could:

`return !!(err && err.message && err.message.match(TIMEOUT_ERROR_MESSAGE));`